src/Stellar/Signer.cpp

```diff
@@ -112,6 +112,33 @@ Data Signer::encode(const Proto::SigningInput& input) const {
             encodeAsset(input.op_change_trust().asset(), data);
             encode64BE(0x7fffffffffffffff, data); // limit MAX
             break;
+
+        case Proto::SigningInput::kOpCreateClaimableBalance:
+            {
+                const auto ClaimantTypeV0 = 0;
+                encodeAsset(input.op_create_claimable_balance().asset(), data);
+                encode64BE(input.op_create_claimable_balance().amount(), data);
+                auto nClaimants = input.op_create_claimable_balance().claimants_size();
+                encode32BE((uint32_t)nClaimants, data);
+                for (auto i = 0; i < nClaimants; ++i) {
+                    encode32BE((uint32_t)ClaimantTypeV0, data);
+                    encodeAddress(Address(input.op_create_claimable_balance().claimants(i).account()), data);
+                    encode32BE((uint32_t)input.op_create_claimable_balance().claimants(i).predicate(), data);
+                    // Note: other predicates not supported, predicate-specific data would follow here
+                }
+            }
+            break;
+
+        case Proto::SigningInput::kOpClaimClaimableBalance:
+            {
+                const auto ClaimableBalanceIdTypeClaimableBalanceIdTypeV0 = 0;
+                encode32BE((uint32_t)ClaimableBalanceIdTypeClaimableBalanceIdTypeV0, data);
+                const auto balanceId = input.op_claim_claimable_balance().balance_id();
+                if (balanceId.size() == 32) {
```

Comment:
Any chance the id size is not 32?

Reply:
Hash ID has no length encoded, 32 is implicit, so if it is different length, returning encoded transaction will be invalid.  That's why I added the check. I refined it now, in that case it returns empty result, and also added test.